### ISSUE TITLE
add `annotations` guide

### ISSUE THREAD

**Issue by g3o2** [vega/vega#782]
Following this [vega-lite related issue](https://github.com/vega/vega-lite/issues/431), I wonder whether it would make sense adding a more general `annotations` guide in the same fashion as the `axes` or `legends` guides, allowing to place a text annotation anywhere in the visualisation. Examples of annotations are title, subtitle, source, explanations within the graph, etc.

Placement could happen in a similar fashion than `legends` (`top`, `right`, `bottom`, `left`, etc. , (x, y) values in encoding block). An array could contain several annotation objects.

Text marks can currently be (mis)used for that purpose but their use as annotations is not obvious to everyone.

**Comment by Fil** [comment]
Yes. Please take a look at the fantastic work on annotations by @susielu
http://d3-annotation.susielu.com/

**Comment by jheer** [comment]
Great feature request. Not sure yet if/when we will get to it, but I agree this could be a very nice addition!

**Comment by domoritz** [comment]
CC @starry97

**Comment by kanitw** [comment]
https://github.com/vega/vega-lite/issues/4055 also ask for subtitle in VL. :)

**Comment by bmschmidt** [comment]
+1. I suspect it's pretty common to want to include 'subtitle' and 'caption'  elements (to use the ggplot2 terms) for explanation and attribution, respectively. Here's another request for subtitles in the altair repo I found searching for this: https://github.com/altair-viz/altair/issues/987.

An annotation grammar would be great, too. But--if you don't mind a maybe uninformed reaction!--other software has led me to think of annotations (which adhere to data inside the plot) and labels (on the margins of the full chart) as distinct, and I'm surprised to see them combined into one feature here. I can't see why subtitles should be 'annotations' in Vega-Lite when--for instance--axis labels are not. Such a general grammar for annotations could be interesting, especially in faceted plots; but I hate to see the capacity to provide basic attribution for data be blocked by something so much more sophisticated. Even the ability to create multiple title elements using the current grammar for a single title would help.

Thanks for your good work.

**Comment by kanitw** [comment]
> But--if you don't mind a maybe uninformed reaction!--other software has led me to think of annotations (which adhere to data inside the plot) and labels (on the margins of the full chart) as distinct, and I'm surprised to see them combined into one feature here. 

That's simply a feature request, not an official plan. :)   If we were to add all/some of these, I think subtitle / caption should be separate from more generic annotations too.

**Comment by jheer** [comment]
I agree that subtitle / caption should be separate. As a starting point, we could consider the design of subtitle specification, perhaps as an extension of the existing `title` directive.

**Comment by g3o2** [comment]
The reason I initially put them together was the grammar of graphics as described in the "literature". Plus, there is a certain fuzziness to all the jargon. Anyhow, theory doesn't always translate 1:1 into practice.

**Comment by kanitw** [comment]
For subtitle and caption, I wonder if we could just extend  the `title` directive to support multiple entries (that are stacked).  With the `orient` and `style` support that we already have, we can easily re-use this mechanism for caption / subtitle / "source" lines. 

(We may want to rename it / provide different syntactic sugar though.)

**Comment by aaronbarzilai** [comment]
Related, I'm interested in chart subtitle as well (or at least wrapping title to two lines). I tried a few different tags, is there a way to force a line break in the title string to simulate a subtitle or second line of a title for now?

**Comment by domoritz** [comment]
There is some related work in https://github.com/vega/vega-lite/pull/3401 and https://github.com/vega/vega/pull/1708.

**Comment by vorbei** [comment]
+1 for 'annotation' channel, it's like the 'tooltip' channel but always visible. I've been doing this manually for repeating encoding same x,y to text marks to do this, but still misses guide lines and other supports. It's a such a needed feature.

**Comment by armsp** [comment]
@groodlooker could you please please elaborate and give some example of what you are talking about. Even I think that being able to add Annotations like the website by @susielu would be great. Since I am new to this I don't understand how you were able to use Vega Embed with d3.

**Comment by kcomplexes** [comment]
@vorbei Do you have any examples of how you have been doing this manually?

Has there been any progress on annotations? is it waiting on [vega/vega-lite#3401](https://github.com/vega/vega-lite/pull/3401), or does it need to be done separately?

**Comment by tbenst** [comment]
I’m not sure if designing syntax is the holdup here or not, so thought I’d point to ggplot2’s seemingly sensible syntax: https://ggplot2.tidyverse.org/reference/annotate.html
